sympy/vector/vector.py

```diff
@@ -250,6 +250,30 @@ def outer(self, other):
 
         return DyadicAdd(*args)
 
+    def vec_project(self, other, scalar=False):
```

Comment:
Nothing big, but since you are enabling both scalar and vector projections with a single function, maybe you could just call this 'projection' or 'project'? I see that you intended to make two functions originally in the PR.